src/engine/controls/cuecontrol.cpp

```diff
@@ -1800,30 +2087,128 @@ void CueControl::hotcueFocusColorNext(double v) {
     pCue->setColor(colorPalette.nextColor(*color));
 }
 
+void CueControl::setCurrentSavedLoopControl(HotcueControl* pControl) {
+    if (m_pCurrentSavedLoopControl && m_pCurrentSavedLoopControl != pControl) {
+        // Disable previous saved loop
+        m_pCurrentSavedLoopControl->setStatus(HotcueControl::Status::Valid);
+        m_pCurrentSavedLoopControl = nullptr;
+    }
+
+    // Set new control as active
+    if (!pControl) {
+        return;
+    }
+
+    QMutexLocker lock(&m_mutex);
+    if (!m_pLoadedTrack) {
+        return;
+    }
+
+    CuePointer pCue(pControl->getCue());
+
+    // Need to unlock before emitting any signals to prevent deadlock.
+    lock.unlock();
+
+    VERIFY_OR_DEBUG_ASSERT(pCue &&
+            pCue->getType() == mixxx::CueType::Loop &&
+            pCue->getEndPosition() != Cue::kNoPosition) {
+        return;
+    }
+
+    m_pCurrentSavedLoopControl = pControl;
+    qDebug() << "CueControl::setLoop" << pCue->getPosition()
+             << pCue->getEndPosition();
+    setLoop(pCue->getPosition(), pCue->getEndPosition(), true);
+    pControl->setStatus(HotcueControl::Status::Active);
+}
+
+void CueControl::slotLoopReset() {
+    setCurrentSavedLoopControl(nullptr);
+}
+
+void CueControl::slotLoopToggled(bool enabled) {
+    if (!m_pCurrentSavedLoopControl) {
+        return;
+    }
+
+    if (enabled) {
+        m_pCurrentSavedLoopControl->setStatus(HotcueControl::Status::Active);
+    } else {
+        m_pCurrentSavedLoopControl->setStatus(HotcueControl::Status::Valid);
+    }
+}
+
+void CueControl::slotLoopUpdated(double startPosition, double endPosition) {
+    if (!m_pCurrentSavedLoopControl) {
+        return;
+    }
+
+    QMutexLocker lock(&m_mutex);
+    if (!m_pLoadedTrack) {
+        return;
+    }
+
+    if (m_pCurrentSavedLoopControl->getStatus() != HotcueControl::Status::Active) {
```

Comment:
Isn't the m_pCurrentSavedLoopControl always active?

Reply:
No. If you activate a saved loop, then disable and reenable the loop (e.g. via reloop_toogle), the saved loop active state should be restored, so it can also be just "valid".

Reply:
OK.